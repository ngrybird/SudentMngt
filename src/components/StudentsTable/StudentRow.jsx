import React from "react";
import Ellipsis from "../../images/Ellipse.png";
class StudentRow extends React.Component {
  render() {
    return (
      <div className="table-row" data-testid="student-row">
        <span>
          <img src={Ellipsis} height="40px" width="40px" />
        </span>
        <span className="row-title">{this.props.name}</span>
        <span className="row-marks">{this.props.marks}</span>
      </div>
    );
  }
}
export default StudentRow;
