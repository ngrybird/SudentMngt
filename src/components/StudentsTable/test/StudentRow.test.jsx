import React from "react";
import { render } from "@testing-library/react";
import StudentRow from "../StudentRow";
import { studentsMarks } from "../../Dashboard/test/testData";

describe("<StudentRow/>", () => {
  it("Verify Students details are rendered", () => {
    const { container, getByText } = render(
      <StudentRow {...studentsMarks[0]} />
    );

    const studentName = getByText("Annete Watson");
    const marks = getByText("9.3");

    expect(studentName).toBeInTheDocument();
    expect(marks).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
