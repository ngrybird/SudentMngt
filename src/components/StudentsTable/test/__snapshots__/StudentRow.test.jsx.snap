// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentRow/> Verify Students details are rendered 1`] = `
<div>
  <div
    class="table-row"
    data-testid="student-row"
  >
    <span>
      <img
        height="40px"
        src="Ellipse.png"
        width="40px"
      />
    </span>
    <span
      class="row-title"
    >
      Annete Watson
    </span>
    <span
      class="row-marks"
    >
      9.3
    </span>
  </div>
</div>
`;
